### PR TITLE
add unflag_first option to yaml flags.

### PR DESCRIPTION
**Pull request by anguta**
add an option to apply_yaml_flags to unflag the uvdata/uvcal/uvflag object before applying flags.

### CODE REVIEW

File: hera_qm/utils.py [hunk 1]
```diff
@@ -722,6 +726,8 @@ def apply_yaml_flags(uv, a_priori_flag_yaml, lat_lon_alt_degrees=None, telescope
         raise NotImplementedError("apply_yaml_flags does not support multiple spws at this time.")
     # if UVCal provided and lst_array is None, get lst_array from times.
     # If lat_lon_alt is not specified, try to infer it from the telescope name, which calfits files generally carry around
+    if unflag_first:
+        uv.flag_array[:] = False
```
Comment (jsdillon) [inline hera_qm/utils.py:730]: I think this is fine for any dimensions, but I'd recommend `uv.flag_array = np.zeros_like(uv.flag_array, dtype=bool)` to be more explicit.

File: hera_qm/tests/test_utils.py [hunk 2]
```diff
@@ -369,7 +363,7 @@ def test_apply_yaml_flags_uvcal(filein, new_metadata):
         uvc.antenna_positions = None
         uvc.lst_array = None
 
-    uvc = utils.apply_yaml_flags(uvc, test_flag)
+    uvc = utils.apply_yaml_flags(uvc, test_flag, unflag_first=True)
```
Comment (jsdillon) [inline hera_qm/tests/test_utils.py:366]: Do you actually want to show that previous flags were removed in this test?

**Review by jsdillon — CHANGES_REQUESTED** [review]
Couple of minor suggestions.

File: hera_qm/utils.py (hunk 1 above)
Comment (anguta) [reply hera_qm/utils.py]: Done.

File: hera_qm/tests/test_utils.py (hunk 2 above)
Comment (anguta) [reply hera_qm/tests/test_utils.py:367]: I've added a check.

**Comment by anguta** [comment]
> Couple of minor suggestions.

Thanks for the feedback. Should be good to go (pending tests passing 🤞 )

**Review by jsdillon — APPROVED** [review]

**Pull request merged**